bin/clover/src/pipeline-steps/generateSubAssets.ts

```diff
@@ -0,0 +1,134 @@
+import { PkgSpec } from "../bindings/PkgSpec.ts";
+import _ from "npm:lodash";
+import {
+  createDefaultProp,
+  ExpandedPropSpec,
+  isExpandedPropSpec,
+} from "../spec/props.ts";
+import { ulid } from "https://deno.land/x/ulid@v0.3.0/mod.ts";
+import { SchemaVariantSpec } from "../bindings/SchemaVariantSpec.ts";
+import { attrFuncInputSpecFromSocket } from "../spec/sockets.ts";
+import { createSocket } from "../spec/sockets.ts";
+import { attrFuncInputSpecFromProp } from "../spec/sockets.ts";
+import { getSiFuncId } from "../spec/siFuncs.ts";
+
+export function generateSubAssets(specs: PkgSpec[]): PkgSpec[] {
+  const newSpecs = [] as PkgSpec[];
+  for (const spec of specs) {
+    const schema = spec.schemas[0];
+    if (!schema) {
+      console.log(
+        `Could not generate default props and sockets for ${spec.name}: missing schema`,
+      );
+    }
+    const schemaVariant = schema.variants[0];
```

Comment:
I assume the zero indexed variant is the default. Is that always true and if not, is that even relevant?

Reply:
similar to above, it's a requirement of the generator to create one schema per spec and a single variant per schema for every variant, and that one will be default.